File: app/src/main/java/com/example/hellokt/application/HelloApplication.java
package com.example.hellokt.application;

import android.app.Application;
import android.content.Context;

import com.example.hellokt.utils.GlobalContext;

public class HelloApplication extends Application {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        GlobalContext.Companion.setAppContext(base);
    }
}

File: app/src/main/java/com/example/hellokt/application/HelloApplication.kt
package com.example.hellokt.application

import android.app.Application
import android.content.Context
import com.alibaba.android.arouter.launcher.ARouter
import com.example.hellokt.utils.GlobalContext

class HelloApplication : Application() {
    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        GlobalContext.setAppContext(base)
        //初始化路由管理
        ARouter.init(this)
    }
}